app/views/RegisterView/index.tsx

```diff
@@ -0,0 +1,350 @@
+		.string()
+		.oneOf([yup.ref('password'), null])
+		.required()
+});
+
+interface IProps extends IBaseScreen<OutsideParamList, 'RegisterView'> {}
+
+const RegisterView = ({ navigation, route }: IProps) => {
+	const dispatch = useDispatch();
+	const { colors } = useTheme();
+	const { Accounts_CustomFields, Site_Url, Accounts_EmailVerification, Accounts_ManuallyApproveNewUsers, showLoginButton } =
+		useAppSelector(state => ({
+			...state.settings,
+			showLoginButton: getShowLoginButton(state),
+			Site_Url: state.settings.Site_Url as string,
+			Accounts_CustomFields: state.settings.Accounts_CustomFields as string
+		}));
+	const {
+		control,
+		handleSubmit,
+		setFocus,
+		getValues,
+		watch,
+		formState: { isValid, isDirty }
+	} = useForm({
+		mode: 'onChange',
+		defaultValues: {
+			name: '',
+			email: '',
+			password: '',
+			confirmPassword: '',
+			username: ''
+		},
+		resolver: yupResolver(validationSchema)
+	});
+	const password = watch('password');
+	const parsedCustomFields = getParsedCustomFields(Accounts_CustomFields);
+	const [customFields, setCustomFields] = useState(getCustomFields(parsedCustomFields));
+	const [saving, setSaving] = useState(false);
+
+	const login = () => {
+		navigation.navigate('LoginView', { title: new parse(Site_Url).hostname });
+	};
+
+	const validateDefaultFormInfo = () => {
+		const isValid = validationSchema.isValidSync(getValues());
+		let requiredCheck = true;
+		Object.keys(parsedCustomFields).forEach((key: string) => {
+			if (parsedCustomFields[key].required) {
+				requiredCheck = requiredCheck && customFields[key] && Boolean(customFields[key].trim());
+			}
+		});
+		return isValid && requiredCheck;
+	};
+
+	const onSubmit = async (data: any) => {
+		logEvent(events.REGISTER_DEFAULT_SIGN_UP);
+
+		const { name, email, password, username } = data;
+
+		if (!validateDefaultFormInfo()) {
+			return;
+		}
+
+		if (!isValidEmail(email)) {
+			showErrorAlert(I18n.t('Invalid_email'), I18n.t('Error'));
+			return;
+		}
+
+		setSaving(true);
+		Keyboard.dismiss();
+
+		try {
+			const response = await Services.register({ name, email, pass: password, username });
+
+			if (response.success) {
+				if (Accounts_EmailVerification) {
+					showErrorAlert(I18n.t('Verify_email_desc'), I18n.t('Registration_Succeeded'));
+					navigation.goBack();
+				} else if (Accounts_ManuallyApproveNewUsers) {
+					showErrorAlert(I18n.t('Wait_activation_warning'), I18n.t('Registration_Succeeded'));
+					navigation.goBack();
+				} else {
+					dispatch(loginRequest({ user: email, password }, false, false, customFields));
+				}
+			}
+		} catch (error: any) {
+			if (error.data?.errorType === 'username-invalid') {
+				return dispatch(loginRequest({ user: email, password }));
+			}
+			if (error.data?.error) {
+				logEvent(events.REGISTER_DEFAULT_SIGN_UP_F);
+				showErrorAlert(error.data.error, I18n.t('Oops'));
+			}
+		} finally {
+			setSaving(false);
+		}
+	};
+
+	const renderCustomFields = () => {
```

Comment:
Create a component to reuse this on ProfileView

Reply:
got it, created.